ops/model.py

```diff
@@ -197,11 +199,12 @@ def __getitem__(self, key):
 class RelationMapping(Mapping):
     """Map of relation names to lists of Relation instances."""
 
-    def __init__(self, relation_names, our_unit, backend, cache):
+    def __init__(self, relations_meta, our_unit, backend, cache):
+        self._relations_meta = relations_meta
```

Comment:
Same here.. we're also changing from getting some data we just iterate over to build the type into storing part of the meta map. Yet, we don't actually use the map much. It'd be faster and nice to use if we did something like this I think:
```
self._peers = set()
for name, data in relations_meta:
    if data.role == 'peers':
        self._peers.add(name)
```
Now `name in self._peers` is all we need.

Reply:
Yes, this feels cleaner, thanks for spotting.